### PR TITLE
BE sending order

### PR DESCRIPTION
**Pull request by lei29**
order model, get order from both chef or customers, post order and chef and customer store the order, put request to change the status of order, like open, accepted, paid, closed... **Currently only got 2 types , open, closed.**

### CODE REVIEW

File: server/routes/order.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+const { Chef,Customer,Order, User } = require("../models/index");
+const express = require("express");
+const router = express.Router();
+const _ = require("lodash");
+const Joi = require("joi");
+const createError = require("http-errors");
+const { decodeToken, userIsAuthorized } = require("../middleware/auth");
+const fileUploadService = require("../services/fileUploader");
```
Comment (smithpg) [inline server/routes/order.js:8]:
```suggestion

```

File: server/routes/order.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * POST to /order/:chef_id/:customer_id to create new order 
+ *  and push order_id to chef's and customer's orders
+ */
+router.post("/:chef_id/:customer_id", async (req, res, next) => {
+    const { error } = validateOrder(req.body);
+    if (error) return next(createError(400, error.details[0].message));
+
+    const order = await Order.create(req.body);
+    const chef = await Chef.findById(req.params.chef_id);
```
Comment (smithpg) [inline server/routes/order.js:29]: As an alternative, we could add a pre('save') hook to Order model similar to the hook that exists on the Dish model for adding new dish to its chef's document

File: server/routes/order.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+const fileUploadService = require("../services/fileUploader");
+
+/**
+ * GET chef's or customer's orders
+ */
+router.get("/:user_id", async (req, res, next) => {
+
+    const user = await User.findById(req.params.user_id);
+
+  res.status(200).send(user.orders);
```
Comment (smithpg) [inline server/routes/order.js:17]: Does this send the actual order objects, or just their id's? I think the default behavior when you retrieve a document that has ObjectIds stored on it is that you need to call populate on the returned model in order to get the other documents referenced by the ids.

Comment (lei29) [reply server/routes/order.js:17]: Got it i will change that

**Pull request merged**